rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/persistence/RedisBackedBakeStore.groovy

```diff
@@ -355,10 +399,34 @@ class RedisBackedBakeStore implements BakeStore {
   }
 
   @Override
-  public boolean deleteBakeByKey(String bakeKey) {
-    def keyList = [bakeKey, "allBakes", thisInstanceIncompleteBakesKey]
+  public String deleteBakeByKey(String bakeKey) {
+    def jedis = jedisPool.getResource()
+    def keyList = [bakeKey, "allBakes"]
+
+    jedis.withCloseable {
+      Set<String> incompleteBakesKeys = jedis.keys(allIncompleteBakesKeyPattern)
+
+      keyList += incompleteBakesKeys
+    }
+
+    return evalSHA("deleteBakeByKeySHA", keyList, [])
+  }
+
+  @Override
+  public String deleteBakeByKeyPreserveDetails(String bakeKey) {
```

Comment:
Bake details aren't typically deleted. Only way (now) is via an explicit deletion call.
One key use case we are trying to enable here is the following:
  - Pipeline containing a bake executes. It results in a Bake execution details link being surfaced in the ui.
  - Later on, the requested image is 'rebaked'.
  - The original Bake execution details link should still function.